consensus/types/src/payload.rs

```diff
@@ -696,7 +696,15 @@ macro_rules! impl_exec_payload_common {
             }
 
             fn is_default_with_empty_roots(&self) -> bool {
-                self.$wrapped_field == $wrapped_type::from($wrapped_type_full::default())
+                // FIXME: is there a better way than ignoring this lint?
+                // This is necessary because the first invocation of this macro might expand to:
+                //     self.execution_payload_header == ExecutionPayloadHeaderMerge::from(ExecutionPayloadMerge::default())
+                // but the second invocation might expand to:
+                //     self.execution_payload == ExecutionPayloadMerge::from(ExecutionPayloadMerge::default())
+                #[allow(clippy::cmp_owned)]
```

Comment:
I suspect the compiler is smart enough to optimise most of this out. There shouldn't be any heap allocations because we only create empty Vecs/lists (which don't allocate) and the most of the checks should boil down to `payload.field == 0`.

I think the only avenue for optimisation on our part here would be to cache the tree hash roots of the empty lists, so they don't get recreated in the `ExecutionPayloadHeader::from(ExecutionPayload)` case. I'm happy to leave that for a future PR though

Reply:
Sounds good! I'll leave the FIXME in there then